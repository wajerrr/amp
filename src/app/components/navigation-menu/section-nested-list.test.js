import React from 'react';
import renderer from 'react-test-renderer';
import NestedAccordionList from './section-nested-list';

describe('NestedAccordionList', () => {
  const nestedSectionList = {
    headline: 'Topics',
    id: '21741809',
    hasPart: {
      parts: [
        {
          headline: 'Topics',
          id: '21741809',
          hasPart: {
            parts: [
              {
                headline: 'Latest updates',
                id: 'storyCollection_21741809_0',
                url: {
                  canonical: 'https://www.economist.com/latest-updates',
                },
              },
            ],
          },
        },
      ],
    },
  };
  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <NestedAccordionList
          title={nestedSectionList.headline}
          list={nestedSectionList}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
