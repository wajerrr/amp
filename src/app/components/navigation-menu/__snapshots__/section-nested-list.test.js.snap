// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NestedAccordionList should match snapshot 1`] = `
<amp-accordion
  id="accordion"
>
  <section
    className={undefined}
  >
    <h4
      className={undefined}
    >
      <span>
        Topics
      </span>
      <span
        className={undefined}
      >
        <svg
          version="1.1"
          viewBox="0 0 48 48"
          x="0px"
          xmlSpace="preserve"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <polygon
            points="33.2,18.6 36,21.4 24,33.4 12,21.4 14.8,18.6 24,27.7 	"
          />
        </svg>
      </span>
    </h4>
    <div>
      <amp-accordion
        id="accordion"
      >
        <section
          className={undefined}
        >
          <h4
            className={undefined}
          >
            <span>
              Topics
            </span>
            <span
              className={undefined}
            >
              <svg
                version="1.1"
                viewBox="0 0 48 48"
                x="0px"
                xmlSpace="preserve"
                xmlns="http://www.w3.org/2000/svg"
                y="0px"
              >
                <polygon
                  points="33.2,18.6 36,21.4 24,33.4 12,21.4 14.8,18.6 24,27.7 	"
                />
              </svg>
            </span>
          </h4>
          <div>
            <a
              className="css-1h5t0lq-StyledLink e2cf1b50"
              href="https://www.economist.com/latest-updates"
            >
              Latest updates
            </a>
          </div>
        </section>
      </amp-accordion>
    </div>
  </section>
</amp-accordion>
`;
